tests/sentry/api/endpoints/test_project_metrics.py

```diff
@@ -0,0 +1,169 @@
+from sentry.testutils import APITestCase
+
+
+class ProjectMetricsTest(APITestCase):
+
+    endpoint = "sentry-api-0-project-metrics-index"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+
+    def test_response(self):
+        response = self.get_valid_response(self.project.organization.slug, self.project.slug)
+
+        required_fields = {"name", "operations", "tags"}
+        optional_fields = {"unit"}
+
+        for item in response.data:
+
+            # All required fields are there:
+            assert required_fields <= item.keys()
+
+            # Only optional field is unit:
+            additional_fields = item.keys() - required_fields
+            if additional_fields:
+                assert additional_fields <= optional_fields
+
+
+class ProjectMetricsTagsTest(APITestCase):
+
+    endpoint = "sentry-api-0-project-metrics-tags"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+
+    def test_unknown_metric(self):
```

Comment:
Would be good to get tests for project permissions too.

Reply:
Done, see https://github.com/getsentry/sentry/pull/25336/commits/134149350914a9af544ad78c620f8e62b8ca96b0.